libbeat/kibana/client.go

```diff
@@ -132,13 +135,26 @@ func NewClientWithConfig(config *ClientConfig) (*Client, error) {
 		return nil, err
 	}
 
+	// check to see if we have a proxy_host set
+	var proxy func(*http.Request) (*url.URL, error)
+	if config.ProxyHost != "" {
+		proxyURL, err := url.Parse(config.ProxyHost)
+		if err != nil {
+			return nil, errors.Wrapf(err, "could not convert %s to url", config.ProxyHost)
+		}
+		proxy = http.ProxyURL(proxyURL)
+	} else if envCfg := httpproxy.FromEnvironment(); envCfg.HTTPProxy != "" || envCfg.HTTPSProxy != "" {
+		proxy = http.ProxyFromEnvironment
+	}
```

Comment:
The httpproxy package looks interesting, as it also takes the NOPROXY environment variable into account. How about relying on `httpproxy` always. E.g.

```
proxyEnv := httpproxy.FromEnvironment()
if config.ProxyURL != nil {
  proxyEnv.HTTPProxy = config.ProxyURL
  proxyEnv.HTTPSProxy = config.ProxyURL
}
proxy := proxyEnv.ProxyFunc()
```